### PR TITLE
Feature/helm http server

### PR DESCRIPTION
**Pull request by jacobcrawford**
Added docker registry and helm chart registry.

To support HA we use data mounts for both the image registry and the helm registry. The pods should mount the same volume for HA.

### CODE REVIEW

**Review by klausenbusk — DISMISSED** [review]
No PVCs?

**Comment by skovbakke** [comment]
> No PVCs?

@klausenbusk, I think you should expand on this. To avoid confusion and improve overall review quality.

**Comment by klausenbusk** [comment]
> @klausenbusk, I think you should expand on this.

AFAIU we don't want the Helm thing and Registry to be volatile. So the data should be persisted? This could be by creating a PVC or perhaps the Registry could use Ceph object storage (S3)?

**Comment by jacobcrawford** [comment]
I agree that we want support for HA for the artifact repository. For the helm registry, there is already the possibility to mount a volume and the specific volume (Ceph in the case of Yggdrasil) should be configured in the values file of Yggdrasil. 
I will add support for the same for the Docker registry.

**Comment by klausenbusk** [comment]
> For the helm registry, there is already the possibility to mount a volume and the specific volume (Ceph in the case of Yggdrasil) should be configured in the values file of Yggdrasil.

Shouldn't the PVCs be part of the chart? If not, what is in charge of creating them?

**Comment by jacobcrawford** [comment]
> > For the helm registry, there is already the possibility to mount a volume and the specific volume (Ceph in the case of Yggdrasil) should be configured in the values file of Yggdrasil.
> 
> Shouldn't the PVCs be part of the chart? If not, what is in charge of creating them?

I think you are right. We need to create the PVCs ourselves as they cannot be created directly from referencing them in a Pod template file and we do not want to use ´cephfs´ volume type.

**Comment by klausenbusk** [comment]
> We need to create the PVCs ourselves as they cannot be created directly from referencing them in a Pod template file and we do not want to use `cephfs` [volume type](https://github.com/kubernetes/examples/tree/master/volumes/cephfs/).

Why don't we want to use `cephfs`?

**Review by skovbakke — DISMISSED** [review]
LGTM

File: chart/values.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      cpu: 100m
+      memory: 128Mi
+    requests:
+      cpu: 100m
+      memory: 128Mi
+  affinity: {}
+  dataVolume:
+    size: 1G
+    storageClass: standard
+    accessMode: ReadWriteOnce
```
Comment (klausenbusk) [inline chart/values.yaml:21]: So we can't do a rolling update?🤔

Comment (jacobcrawford) [reply chart/values.yaml:21]: In production (Yggdrasil) we will use Ceph's storageClass which supports other accessModes. This is just a configuration needed when we don't have more sophisticated storage solutions.

**Comment by jacobcrawford** [comment]
Moved to mono-repo

**Pull request closed without merging**